File: app/src/main/java/at/ac/tuwien/caa/docscan/ui/BaseNoNavigationActivity.java
package at.ac.tuwien.caa.docscan.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import android.view.MenuItem;

import at.ac.tuwien.caa.docscan.R;

/**
 * Created by fabian on 25.08.2017.
 */

public abstract class BaseNoNavigationActivity extends AppCompatActivity {

    /**
     * Sets the title of the toolbar.
     * @param titleID ID of the title string defined in strings.xml
     */
    protected void initToolbarTitle(int titleID) {

        Toolbar toolbar = findViewById(R.id.main_toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setTitle(getString(titleID));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

    }

    /**
     * Sets the title of the toolbar.
     * @param title title string
     */
    protected void initToolbarTitle(String title) {

        Toolbar toolbar = findViewById(R.id.main_toolbar);
        setSupportActionBar(toolbar);
        if (title != null)
            getSupportActionBar().setTitle(title);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()){
            case android.R.id.home:
                onBackPressed();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/at/ac/tuwien/caa/docscan/ui/base/BaseNoNavigationActivity.kt
package at.ac.tuwien.caa.docscan.ui.base

import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import at.ac.tuwien.caa.docscan.R

/**
 * Created by fabian on 25.08.2017.
 */
abstract class BaseNoNavigationActivity : BaseActivity() {

    /**
     * Sets the title of the toolbar.
     *
     * @param titleID ID of the title string defined in strings.xml
     */
    protected fun initWithTitle(titleID: Int?) {
        val toolbar = findViewById<Toolbar>(R.id.main_toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = if(titleID != null) getString(titleID) else ""
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
